pygsti/models/modelconstruction.py

```diff
@@ -903,7 +903,15 @@ def _spec_to_densevec(spec, is_prep):
                     num_qudits = len(qudit_labels)
                     if isinstance(spec, str):
                         if spec.isdigit():  # all([l in ('0', '1') for l in spec]): for qubits
-                            bydigit_index = effect_spec
+                            bydigit_index = spec
+                            assert (len(bydigit_index) == num_qudits), \
+                                "Wrong number of qudits in '%s': expected %d" % (spec, num_qudits)
+                            v = _np.zeros(state_space.udim)
+                            inc = _np.flip(_np.cumprod(list(reversed(processor_spec.qudit_udims[1:] + (1,)))))
```

Comment:
Probably not worth changing since Instruments + Pspec + model construction needs a bit of an overhaul, but it's not immediately clear what this is doing. My best guess is pulling out the effect index in the case where we have MCMs and our POVM labels look something like ('0', '1') or something like that? A comment here would be super appreciated, but not PR blocking.

Reply:
I'm not 100% confident on this (it looks like I copied and pasted this code from existing POVM parsing logic since the effect logic was the same), but I think what this is doing is converting a string of digits (possibly corresponding to a mixed-base) integer into a base-10 integer. This is probably fragile against anything larger than a qunon.

Reply:
Comment added

Reply:
Ty!